php-mode.el

```diff
@@ -1734,6 +1734,13 @@ The output will appear in the buffer *PHP*."
 (dolist (pattern '("\\.php[s345t]?\\'" "/\\.php_cs\\(\\.dist\\)?\\'" "\\.phtml\\'" "/Amkfile\\'" "\\.amk\\'"))
   (add-to-list 'auto-mode-alist `(,pattern . php-mode) t))
 
+;;;###autoload
+(progn
+  ;; matches `/config.php'
+  ;; refs https://github.com/ejmr/php-mode/pull/362#issuecomment-312404436
+  ;; To avoid conflict with default `conf-maybe-mode' in auto-mode-alist, add it explicitly.
+  (add-to-list 'auto-mode-alist '("[/.]c\\(?:on\\)?f\\(?:i?g\\)?\\(?:\\.[a-zA-Z0-9._-]+\\)?\\.php[s345t]?\\'" . php-mode)))
+
```

Comment:
This pattern is This pattern is a modification of [copying from `files.el`](https://github.com/emacs-mirror/emacs/blob/65eee8392ff95f58f7b0bd036e1fe065523658c6/lisp/files.el#L2620).

```el
;; Original code: https://github.com/emacs-mirror/emacs/blob/65eee8392ff95f58f7b0bd036e1fe065523658c6/lisp/files.el#L2620
     ("[/.]c\\(?:on\\)?f\\(?:i?g\\)?\\(?:\\.[a-zA-Z0-9._-]+\\)?\\'" . conf-mode-maybe)
```

I think that rewriting this regexp pattern to `rx` is not appropriate.